### PR TITLE
Handle successful disputes & automatically create payouts

### PR DESCRIPTION
**Pull request by sampoder**
Closes https://github.com/hackclub/hcb/issues/2641. These won't be instant but they should now be automatic.

### CODE REVIEW

File: app/controllers/stripe_controller.rb [hunk 1]
```diff
@@ -298,4 +298,23 @@ def handle_issuing_personalization_design_updated(event)
   alias_method :handle_issuing_personalization_design_rejected, :handle_issuing_personalization_design_updated
   alias_method :handle_issuing_personalization_design_deactivated, :handle_issuing_personalization_design_updated
 
+  def handle_issuing_dispute_funds_reinstated
+    dispute = event.data.object
+    transaction = Stripe::Issuing::Transaction.retrieve(dispute["transaction"])
+    hcb_code = RawPendingStripeTransaction.find_or_initialize_by(stripe_transaction_id: transaction["authorization"]).canonical_pending_transaction.local_hcb_code
```
Comment (garyhtou) [inline app/controllers/stripe_controller.rb]:
```suggestion
    hcb_code = RawPendingStripeTransaction.find_by!(stripe_transaction_id: transaction["authorization"]).canonical_pending_transaction.local_hcb_code
```

File: app/controllers/stripe_controller.rb [hunk 2]
```diff
@@ -298,4 +298,23 @@ def handle_issuing_personalization_design_updated(event)
...
+        currency: dispute["currency"],
+        statement_descriptor: "HCB-#{hcb_code.short_code}",
+        source_balance: "issuing",
+        metadata: {
+          dispute: dispute["id"],
+          authorization: transaction["authorization"],
+          hcb_code: hcb_code.hcb_code
+        }
+      )
+    end
```
Comment (garyhtou) [inline app/controllers/stripe_controller.rb]: airbrake if we get a different state? i don't think we could

File: app/controllers/stripe_controller.rb [hunk 3]
```diff
@@ -298,4 +298,23 @@ def handle_issuing_personalization_design_updated(event)
...
+    transaction = Stripe::Issuing::Transaction.retrieve(dispute["transaction"])
+    hcb_code = RawPendingStripeTransaction.find_or_initialize_by(stripe_transaction_id: transaction["authorization"]).canonical_pending_transaction.local_hcb_code
+    if dispute["status"] == "won"
+      StripeService::Payout.create(
+        amount: dispute["amount"],
+        currency: dispute["currency"],
+        statement_descriptor: "HCB-#{hcb_code.short_code}",
+        source_balance: "issuing",
+        metadata: {
+          dispute: dispute["id"],
```
Comment (garyhtou) [inline app/controllers/stripe_controller.rb]:
```suggestion
          dispute_id: dispute["id"],
```

File: app/controllers/stripe_controller.rb [hunk 4]
```diff
@@ -298,4 +298,23 @@ def handle_issuing_personalization_design_updated(event)
...
   alias_method :handle_issuing_personalization_design_deactivated, :handle_issuing_personalization_design_updated
 
+  def handle_issuing_dispute_funds_reinstated
+    dispute = event.data.object
+    transaction = Stripe::Issuing::Transaction.retrieve(dispute["transaction"])
+    hcb_code = RawPendingStripeTransaction.find_or_initialize_by(stripe_transaction_id: transaction["authorization"]).canonical_pending_transaction.local_hcb_code
+    if dispute["status"] == "won"
+      StripeService::Payout.create(
+        amount: dispute["amount"],
+        currency: dispute["currency"],
```
Comment (garyhtou) [inline app/controllers/stripe_controller.rb]: sanity check currency is USD when for foreign charges

File: app/controllers/stripe_controller.rb [hunk 5]
```diff
@@ -298,4 +298,23 @@ def handle_issuing_personalization_design_updated(event)
   alias_method :handle_issuing_personalization_design_rejected, :handle_issuing_personalization_design_updated
   alias_method :handle_issuing_personalization_design_deactivated, :handle_issuing_personalization_design_updated
 
+  def handle_issuing_dispute_funds_reinstated
+    dispute = event.data.object
+    transaction = Stripe::Issuing::Transaction.retrieve(dispute["transaction"])
+    hcb_code = RawPendingStripeTransaction.find_or_initialize_by(stripe_transaction_id: transaction["authorization"]).canonical_pending_transaction.local_hcb_code
+    if dispute["status"] == "won"
```
Comment (garyhtou) [inline app/controllers/stripe_controller.rb]: raise & airbrake if non-USD

File: app/controllers/stripe_controller.rb [hunk 6]
```diff
@@ -298,4 +298,23 @@ def handle_issuing_personalization_design_updated(event)
...
+      StripeService::Payout.create(
+        amount: dispute["amount"],
+        currency: dispute["currency"],
+        statement_descriptor: "HCB-#{hcb_code.short_code}",
+        source_balance: "issuing",
+        metadata: {
+          dispute: dispute["id"],
+          authorization: transaction["authorization"],
```
Comment (garyhtou) [inline app/controllers/stripe_controller.rb]:
```suggestion
          authorization_id: transaction["authorization"],
```

File: app/controllers/stripe_controller.rb (hunk 5 above)
Comment (sampoder) [reply app/controllers/stripe_controller.rb]:
```suggestion
    if dispute["status"] == "won" && dispute["currency"] == "USD"
```

**Pull request merged**